tools/graph-info/graph-info.cpp

```diff
@@ -0,0 +1,147 @@
+      break;
+    }
+    }
+  }
+}
+// We could find out this information without loading the graph,
+// but that would take more effort
+void
+PrintDist(const std::string& src_uri, int this_host, int num_hosts) {
+  auto res = katana::PropertyFileGraph::Make(src_uri);
+  if (!res) {
+    KATANA_LOG_FATAL("error loading graph: {}", res.error());
+  }
+  auto g = std::move(res.value());
+
+  fmt::print("{:16} : {:2} : {:>7}\n", "Nodes", this_host, g->num_nodes());
+  fmt::print("{:16} : {:2} : {:>7}\n", "Edges", this_host, g->num_edges());
+
+  auto masters_range = g->masters();
+  auto outgoing_mirrors_range = g->outgoing_mirrors();
+  auto incoming_mirrors_range = g->incoming_mirrors();
+  fmt::print(
+      "{:16} : {:2} : {:>7}\n", "Masters", this_host,
+      std::distance(masters_range.begin(), masters_range.end()));
+  fmt::print(
+      "{:16} : {:2} : {:>7}\n", "Outgoing mirrors", this_host,
+      std::distance(
+          outgoing_mirrors_range.begin(), outgoing_mirrors_range.end()));
+  fmt::print(
+      "{:16} : {:2} : {:>7}\n", "Incoming mirrors", this_host,
+      std::distance(
+          incoming_mirrors_range.begin(), incoming_mirrors_range.end()));
+
+  if (this_host == (num_hosts - 1)) {
+    fmt::print(
+        "{:16} : {:>2}\n", "Node Properties", g->GetNodePropertyNames().size());
+    PrintNames(g->node_schema(), g->GetNodePropertyNames());
+
+    fmt::print(
+        "{:16} : {:>2}\n", "Edge Properties", g->GetEdgePropertyNames().size());
+    PrintNames(g->edge_schema(), g->GetEdgePropertyNames());
+  }
+}
+
+tsuba::RDGPrefix
+OpenPrefix(const std::string& src_uri) {
+  auto open_res = tsuba::Open(src_uri, tsuba::kReadOnly);
+  if (!open_res) {
+    KATANA_LOG_FATAL("Open RDG failed: {}", open_res.error());
+  }
+  auto prefix_res = tsuba::RDGPrefix::Make(open_res.value());
+  if (!prefix_res) {
+    KATANA_LOG_FATAL("RDGPrefix make failed: {}", prefix_res.error());
+  }
+  return std::move(prefix_res.value());
+}
+
+int
+main(int argc, char** argv) {
+  llvm::cl::ParseCommandLineOptions(argc, argv);
+  // build this after CLI parsing since it might cause MPI to fail
+  auto dist_mem_sys = std::make_unique<katana::DistMemSys>();
+  auto& net = katana::getSystemNetworkInterface();
+  auto num_hosts = net.Num;
+  auto this_host = net.ID;
+
+  if (this_host == 0) {
+    fmt::print("{:16} : {}\n", "URI", src_uri);
+  }
+
+  if (num_hosts == 1) {
+    auto prefix = OpenPrefix(src_uri);
+    if (this_host == 0) {
+      fmt::print("{:16} : {}\n", "Nodes", prefix.num_nodes());
+      fmt::print("{:16} : {}\n", "Edges", prefix.num_edges());
+      fmt::print("{:16} : {}\n", "Version", prefix.version());
+    }
+  } else {
+    PrintDist(src_uri, this_host, num_hosts);
+  }
+  // dist_mem_sys->Fini()  prints Stat output, so do our own shutdown
+  if (auto fini_good = tsuba::Fini(); !fini_good) {
+    KATANA_LOG_ERROR("tsuba::Fini: {}", fini_good.error());
+  }
+  katana::internal::destroySystemNetworkInterface();
```

Comment:
- destroying `dist_mem_sys` should handle this case for you too.
- I'm missing why you wouldn't just call `dist_mem_sys->Fini()` directly as it is supposed to handle the cleanup of everything it initializes (tsuba, katana::getSystemNetworkInterface).

Reply:
I wrote a comment.
`  // dist_mem_sys->Fini()  prints Stat output, so do our own shutdown`
Maybe I'm too sensitive to aesthetics, but the banner output from the Stats printing was confusing and untidy.

Reply:
I see. I saw the comment but I didn't realize it was related to omitting Stats printing altogether.

My suggestion is to file an issue to remove/make optional Stats printing throughout. And either use `dist_mem_sys->Fini()`, and deal with the ugly output in the interim or keep the code referencing internal APIs and be prepared to fix things if there is a change to initialization behavior.

Reply:
Ok, that makes sense.

Reply:
I added an optional flag to Fini for stats printing.  I know Boolean parameters are often not ideal.  I made it optional, which I hope does not violate our coding standards.